### DISCUSSION TITLE
undirectional relationships

### DISCUSSION THREAD

**Discussion by blevine** [prisma/prisma#8857] [Q&A] [answered]
Is there any way to model a unidirectional relationship in Prisma?  My use-case:

I'm accessing a database that belongs to another application.  So I don't "own" that DB schema. I can add tables, but I can't modify existing tables. I want to extend the definition of a User from that application's schema.  My first thought was just to create a one-to-one relations between MyUser and ApplicationUser. But Prisma requires there to be an "opposite relation field" on ApplicationUser which I can't create.  How can I model this in Prisma?

I suppose one way would be to create a join (pivot) table with  applicationUserId and myUserId columns, but I'm hoping there's some other way.

**Comment by ryands17** [comment, marked as answer]
The opposite relation isn't created, it's a virtual relation that helps making queries easier. 1-1 relations are usually created like this:

```prisma
model User {
  id      Int      @id @default(autoincrement())
  profile Profile?
}

model Profile {
  id     Int  @id @default(autoincrement())
  user   User @relation(fields: [userId], references: [id])
  userId Int // relation scalar field (used in the `@relation` attribute above)
}
```

Here, the opposite relation `profile` is a virtual relation and it isn't created in the database so you don't need to worry about that as it won't be created. In the above example, only the `userId` will be stored in the `Profile` table as a foreign key.

**Reply by blevine** [reply]
Ah I see.  Thanks for the explanation

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
